orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/ApplySourceServerGroupSnapshotTaskSpec.groovy

```diff
@@ -0,0 +1,171 @@
+    given:
+    def parentPipeline = new Pipeline()
+    parentPipeline.stages << new PipelineStage(parentPipeline, "", [
+      strategy                         : "custom",
+      sourceServerGroupCapacitySnapshot: [
+        min    : 0,
+        desired: 5,
+        max    : 10
+      ]
+    ])
+    parentPipeline.stages << new PipelineStage(parentPipeline, "", [
+      executionId: "execution-id"
+    ])
+    ((AbstractStage) parentPipeline.stages[0]).id = "stage-1"
+    ((AbstractStage) parentPipeline.stages[1]).parentStageId = "stage-1"
+    ((AbstractStage) parentPipeline.stages[1]).type = "pipeline"
+
+    def childPipeline = new Pipeline()
+    childPipeline.stages << new PipelineStage(childPipeline, "", [
+      type: "doSomething"
+    ])
+    childPipeline.stages << new PipelineStage(childPipeline, "", [
+      type: "createServerGroup"
+    ])
+
+    (parentPipeline.stages + childPipeline.stages).each {
+      ((AbstractStage) it).setStageNavigator(stageNavigator)
+    }
+
+    when:
+    def ancestorDeployStage = ApplySourceServerGroupCapacityTask.getAncestorDeployStage(
+      executionRepository, parentPipeline.stages[-1]
+    )
+
+    then:
+    1 * executionRepository.retrievePipeline("execution-id") >> childPipeline
+
+    ancestorDeployStage == childPipeline.stages[1]
+  }
+
+  @Unroll
+  void "should support ancestor deploy stages w/ a #strategy strategy"() {
+    given:
+    def stage = new PipelineStage(new Pipeline(), "", [
+      strategy                         : strategy,
+      sourceServerGroupCapacitySnapshot: [
+        min    : 0,
+        desired: 5,
+        max    : 10
+      ]
+    ])
+    ((AbstractStage) stage).setStageNavigator(stageNavigator)
+
+    expect:
+    ApplySourceServerGroupCapacityTask.getAncestorDeployStage(null, stage) == stage
+
+    where:
+    strategy   || _
+    "redblack" || _
+    null       || _
+  }
+
+  void "should construct resizeServerGroup context with source `min` + target `desired` and `max` capacity"() {
+    given:
+    def pipeline = new Pipeline()
+    pipeline.stages << new PipelineStage(pipeline, "", [
+      "deploy.server.groups"           : [
+        "us-east-1": ["application-stack-v001"]
+      ],
+      application                      : "application",
+      stack                            : "stack",
+      account                          : "test",
+      region                           : "us-east-1",
+      sourceServerGroupCapacitySnapshot: [
+        min    : 0,
+        desired: 10,
+        max    : 20
+      ]
+    ])
+    pipeline.stages << new PipelineStage(pipeline, "", [account: "test"])
+    pipeline.stages.each {
+      ((AbstractStage) it).setStageNavigator(stageNavigator)
+    }
+
+    ((AbstractStage) pipeline.stages[0]).id = "stage-1"
+    ((AbstractStage) pipeline.stages[1]).parentStageId = "stage-1"
+
+    and:
+    def currentServerGroupResponse = task.objectMapper.writeValueAsString([
```

Comment:
if there's an oortHelper equivalent it will clean up this test (gets rid of the need to serialize the map and construct Response in here)